doc/source/ray-core/objects/limiting-memory-usage.rst

```diff
@@ -0,0 +1,274 @@
+
+Memory Monitor
+--------------
+
+The memory monitor is a component that runs on each node. It periodically checks the memory usage, which includes the worker heap, the object store, and the raylet as described in :ref:`memory management <memory>`. If the combined usage exceeds a configurable threshold the raylet will kill task or actor to free up memory and prevent Ray from failing.
+
+.. note::
+
+    The memory monitor is in :ref:`alpha <api-stability-alpha>`. It is disabled by default and needs to be enabled by setting the environment variable ``RAY_memory_monitor_interval_ms`` to a value greater than zero when Ray starts. It is available on Linux and is tested with Ray running inside a container that is using cgroup v1. If you encounter issues when running the memory monitor outside of a container or the container is using cgroup v2, please :ref:`file an issue or post a question <limiting-memory-usage-questions>`. 
+
+Enabling the Memory Monitor
+---------------------------
+
+Verify the memory monitor is disabled by default:
+
+.. code-block:: bash
+
+    ray stop; ray start --head
+
+The raylet logs will contain a line that tells us whether the monitor is running or not:
+
+.. code-block:: bash
+
+    grep memory_monitor.cc /tmp/ray/session_latest/logs/raylet.out
+
+Which should print
+
+.. code-block:: bash
+
+    (raylet) memory_monitor.cc:65: MemoryMonitor disabled. Specify `memory_monitor_interval_ms` > 0 to enable the monitor.
+
+Now enable the memory monitor:
+
+.. code-block:: bash
+
+    ray stop; RAY_memory_monitor_interval_ms=250 ray start --head
+
+Check the logs again to see the monitor is now running:
+
+.. code-block:: bash
+
+    grep memory_monitor.cc /tmp/ray/session_latest/logs/raylet.out
+
+Which should print
+
+.. code-block:: bash
+
+    (raylet) memory_monitor.cc:45: MemoryMonitor initialized with usage threshold at 34664513536 bytes (0.97 system memory), total system memory bytes: 35738255360
+
+Memory usage threshold
+----------------------
+
+The memory usage threshold is used by the memory monitor to determine when it should start killing processes to free up memory. The threshold is controlled by two environment variables ``RAY_memory_usage_threshold_fraction`` (default: 0.9) and ``RAY_min_memory_free_bytes`` (default: 1 GiB). When the node starts it computes the usage threshold as follows:
+
+.. code-block:: bash
+
+    usage_threshold = max(system_memory * RAY_memory_usage_threshold_fraction, system_memory - RAY_min_memory_free_bytes)
+
+``RAY_min_memory_free_bytes`` can be disabled by setting its value to -1. In that case it only uses ``RAY_memory_usage_threshold_fraction`` to determine the usage threshold. Here we set the memory threshold to be 0.9 of system memory:
+
+.. code-block:: bash
```

Comment:
Thanks for adding these examples, but I'm not sure that it is the best use of space (should err on the side of adding fewer more concise docs if possible). Maybe you could just explain when to override which env variable?